mungegithub/github/github.go

```diff
@@ -1228,6 +1228,12 @@ func (obj *MungeObject) doWaitStatus(pending bool, requiredContexts []string, c
 			glog.V(4).Infof("PR# %d is pending, waiting for %f seconds", *obj.Issue.Number, sleepTime.Seconds())
 		}
 		time.Sleep(sleepTime)
+
+		// If it has been closed, assume that we want to break from the poll loop early.
+		obj.Refresh()
```

Comment:
~4 API calls per minute, which is pretty tiny when the SQ regularly uses 5 API calls per second.